CMakePresets.json

```diff
@@ -0,0 +1,77 @@
+{
+  "version": 3,
+  "configurePresets": [
+    {
+      "name": "base",
+      "hidden": true,
+      "binaryDir": "${sourceDir}/build",
+      "cacheVariables": {
+        "BUILD_SHARED_LIBS": "ON"
+      }
+    },
+    {
+      "name": "production",
+      "inherits": "base",
+      "displayName": "Production release",
+      "description": "Default release configuration as preCICE is intended to be used in production.",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Release",
+        "CMAKE_INTERPROCEDURAL_OPTIMIZATION": "ON"
+      }
+    },
+    {
+      "name": "develop",
+      "inherits": "base",
+      "displayName": "Development config",
+      "description": "Development config with default settings",
+      "cacheVariables": {
+        "CMAKE_BUILD_TYPE": "Debug",
+        "CMAKE_CXX_FLAGS": "-Wall -Wextra -Wno-unused-parameter"
+      },
+      "warnings": {
+        "dev": true,
+        "deprecated": true,
+        "uninitialized": true
+      }
+    },
+    {
+      "name": "debpkg",
+      "inherits": "production",
+      "displayName": "Debian package",
+      "description": "Build for the generation of Debian packages",
+      "cacheVariables": {
+        "CMAKE_INSTALL_PREFIX": "/usr",
+        "CPACK_GENERATOR": "DEB",
+        "BUILD_TESTING": "OFF"
+      }
+    },
+    {
+      "name": "bigrelease",
+      "inherits": "production",
+      "displayName": "Big release",
+      "description": "Release build with log and assertions enabled",
```

Comment:
This term is not self-descriptive to me.
Why not `release-with-assertions`?

Since the presets are only used once in the installation, I think that long, descriptive names are fine.

Reply:
So, this is a release + debug log + assertions.
Not sure how to call this.

Reply:
For me, this would be equivalent to `CMake`s `RelWithDebInfo`. I would go with something similar, maybe `RelWithDebChecks` or so.

Reply:
Based on suggestion:
* verbose-release-with-assertions
* checked-release-with-debug-log
* checked-verbose-release
* debug-in-production
* production-debug

I think that we should add a preset for production + assertions (`checked-production`, `production-audit`, `production-with-assertion`)

Reply:
Or optimized debug